travis.sh

```diff
@@ -141,8 +142,10 @@ my_travis_wait 60 catkin build --no-status --summarize || exit 1
 # Source the new built workspace
 travis_run source install/setup.bash;
 
-# Only run tests on the current repo's packages
-TEST_PKGS=$(catkin_topological_order $CI_SOURCE_PATH --only-names)
+# Choose which packages to run tests on
+echo "Test blacklist: $TEST_BLACKLIST"
+TEST_PKGS=$(catkin_topological_order src --only-names | grep -Fvxf <(echo "$TEST_BLACKLIST" | tr ' ;,' '\n'))
```

Comment:
One thing to note here (my fault really) is that line 147 is dependent on the working directory. Changing to `catkin_topological_order "~/ros/ws_$REPOSITORY_NAME/src" --only-names` makes it more robust against moving the code around in the file.

Part of the file is operating inside the `src` map, part in the parent directory, which is easy to miss (and might take up to 50 minutes or more to find out).